lib/posix/sched.c

```diff
@@ -41,3 +42,57 @@ int sched_get_priority_max(int policy)
 	errno = EINVAL;
 	return -1;
 }
+
+/**
+ * @brief Get scheduling parameters
+ *
+ * See IEEE 1003.1
+ */
+int sched_getparam(pid_t pid, struct sched_param *param)
+{
+	/* TODO: Ensure the Zephyr POSIX process id (PID) value is assigned
+	 * with its first thread's id (TID).
+	 */
+	struct sched_param dummy_param = {0};
+	pthread_t tid = (pthread_t)pid;
```

Comment:
Ah, I see - this is one area that is a bit touchy.

I'm not sure if it's a good idea to mix pid_t and pthread_t

Reply:
yeah, it would be better to have a separate PR first to introduce `pid_t` and enable all these `pid_t` parametrized POSIX APIs, with some macro or inline functions for PID<->TID conversion based on decision for how PID value originates/relates to TID.

Reply:
@golowanow - do you feel that PID and TID are related?